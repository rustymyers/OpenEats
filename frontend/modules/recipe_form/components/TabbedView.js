import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { Popover, OverlayTrigger } from 'react-bootstrap'
import {
  injectIntl,
  defineMessages,
} from 'react-intl';

class TabbedView extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      display: this.props.display || 0,
    };
  }

  switchView = val => {
    this.setState({ display: val })
  };

  render() {
    const { formatMessage } = this.props.intl;
    const messages = defineMessages({
      preview: {
        id: 'recipe.create.preview',
        description: 'Preview',
        defaultMessage: 'Preview',
      },
      editor: {
        id: 'recipe.create.editor',
        description: 'Editor',
        defaultMessage: 'Editor',
      },
    });

    let { children, label, infoTitle, infoDesc, errors } = this.props;
    let { display } = this.state;

    let contentClassName = classNames({
      'content': true,
      'has-error': !!this.props.errors
    });

    let navClassName = classNames({
      'nav': true,
      'nav-tabs': true,
      'has-error': !!this.props.errors
    });

    const popoverHoverFocus = (
      <Popover id="help" title={ infoTitle }>
        { infoDesc }
      </Popover>
    );

    return (
      <div className="live-editor">
        <ul className={ navClassName }>
          <li className={ classNames({ left: true, active: true })}>
            <span>
              { label }
              &nbsp;
              <OverlayTrigger
                  trigger={['hover']}
                  placement="top"
                  overlay={ popoverHoverFocus }
              >
                <span className="glyphicon glyphicon-info-sign"/>
              </OverlayTrigger>
            </span>
          </li>
          <li className={ classNames({ active: display === 0 })}>
            <span onClick={ this.switchView.bind(this, 0) }>
              { formatMessage(messages.editor) }
            </span>
          </li>
          <li className={ classNames({ active: display === 1 })}>
            <span onClick={ this.switchView.bind(this, 1) }>
              { formatMessage(messages.preview) }
            </span>
          </li>
        </ul>
        <div className={ contentClassName }>{ children[display] }</div>
        <div className="help-inline">{ errors }</div>
      </div>
    )
  }
}

TabbedView.PropTypes = {
  label: PropTypes.string.isRequired,
  display: PropTypes.number,
  errors: PropTypes.string,
  className: PropTypes.string,
  infoTitle: PropTypes.string,
  infoDesc: PropTypes.string,
  intl: PropTypes.obj,
};

export default injectIntl(TabbedView)
