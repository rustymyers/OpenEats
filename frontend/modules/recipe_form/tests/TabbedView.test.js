import React from 'react';
import TabbedView from '../components/TabbedView';
import createComponentWithIntl from '../../../jest_mocks/createComponentWithIntl';

test('TabbedView component test', () => {
  const component = createComponentWithIntl(
    <TabbedView>
      hi
      hi
    </TabbedView>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
