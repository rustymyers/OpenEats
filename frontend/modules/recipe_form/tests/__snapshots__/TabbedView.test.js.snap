// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabbedView component test 1`] = `
<div
  className="live-editor"
>
  <ul
    className="nav nav-tabs"
  >
    <li
      className="left active"
    >
      <span>
        Â 
        <span
          className="glyphicon glyphicon-info-sign"
          onClick={null}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        />
      </span>
    </li>
    <li
      className="active"
    >
      <span
        onClick={[Function]}
      >
        Editor
      </span>
    </li>
    <li
      className=""
    >
      <span
        onClick={[Function]}
      >
        Preview
      </span>
    </li>
  </ul>
  <div
    className="content"
  >
    h
  </div>
  <div
    className="help-inline"
  />
</div>
`;
